core/src/test/java/io/questdb/griffin/AlterTableSetTypeRestartTest.java

```diff
@@ -0,0 +1,384 @@
+            }
+            validateShutdown();
+
+            // restart
+            try (final ServerMain questdb = new ServerMain("-d", root.toString(), Bootstrap.SWITCH_USE_DEFAULT_LOG_FACTORY_CONFIGURATION)) {
+                questdb.start();
+
+                final CairoEngine engine = questdb.getCairoEngine();
+                try {
+                    engine.getTableToken(tableName);
+                } catch (CairoException e) {
+                    assertTrue(e.getMessage().contains("[-1] table does not exist [table=testtable]"));
+                }
+            }
+        });
+    }
+
+    private static void assertConvertFileContent(Path convertFilePath, byte expected) throws IOException {
+        final byte[] fileContent = java.nio.file.Files.readAllBytes(java.nio.file.Paths.get(convertFilePath.toString()));
+        assertEquals(1, fileContent.length);
+        assertEquals(expected, fileContent[0]);
+    }
+
+    private static void assertConvertFileDoesNotExist(CairoEngine engine, String tableName) {
+        doesConvertFileExist(engine, tableName, false);
+    }
+
+    private static Path assertConvertFileExists(CairoEngine engine, String tableName) {
+        return doesConvertFileExist(engine, tableName, true);
+    }
+
+    private static void assertNumOfRows(CairoEngine engine, String tableName, int count) throws SqlException {
+        try (
+                final SqlExecutionContext context = createSqlExecutionContext(engine);
+                final SqlCompiler compiler = new SqlCompiler(engine)
+        ) {
+            TestUtils.assertSql(
+                    compiler,
+                    context,
+                    "SELECT count() FROM " + tableName,
+                    Misc.getThreadLocalBuilder(),
+                    "count\n" +
+                            count + "\n"
+            );
+        }
+    }
+
+    private static SqlExecutionContext createSqlExecutionContext(CairoEngine engine) {
+        return new SqlExecutionContextImpl(engine, 1).with(
+                AllowAllCairoSecurityContext.INSTANCE,
+                null,
+                null,
+                -1,
+                null
+        );
+    }
+
+    private static void createTable(String tableName, String walMode) throws SQLException {
+        runSqlViaPG("create table " + tableName + " (ts timestamp, x long) timestamp(ts) PARTITION BY DAY " + walMode);
+        LOG.info().$("created table: ").utf8(tableName).$();
+    }
+
+    private static Path doesConvertFileExist(CairoEngine engine, String tableName, boolean doesExist) {
+        final TableToken token = engine.getTableToken(tableName);
+        final Path path = Path.PATH.get().of(engine.getConfiguration().getRoot()).concat(token).concat(WalUtils.CONVERT_FILE_NAME);
+        MatcherAssert.assertThat(Chars.toString(path), Files.exists(path.$()), Matchers.is(doesExist));
+        return doesExist ? path : null;
+    }
+
+    private static void drainWalQueue(CairoEngine engine) {
```

Comment:
could be protected and in the parent class.
I have a test that relies waiting for a query to complete, to ensure the WAL has been applied, I would use this method instead, I thought of it.

Reply:
fixed in [509c5d5](https://github.com/questdb/questdb/pull/2947/commits/509c5d5677a0aa8f5bac45ee8878683fc16321fe)

Reply:
thank you!